### PR TITLE
fix: fix save file bug

### LINKED ISSUE #585: Remove the last line breaker when SELECT INTO csv files
Now we will add extra the line breaker at the end of output CSV file. We should fix like what Mysql/Spark/... does.

### PR DESCRIPTION
**Pull request by zekai427**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Review by tobegit3hub — CHANGES_REQUESTED** [review]
Fix the warning from cpplint.

File: src/cmd/sql_cmd_test.cc [hunk 1]
```diff
@@ -103,13 +103,13 @@ TEST_F(SqlCmdTest, select_into_outfile) {
...
-    std::string data;
-    while (!file.eof()) {
-        getline(file, line);
-        data.append(line);
-    }
-    ASSERT_EQ(data, "col1,col2key1,1");
+    file.seekg (0, file.end);
+    int length = file.tellg();
+    file.seekg (0, file.beg);
+    char* data = new char[length+1];
```
Comment (tobegit3hub) [inline src/cmd/sql_cmd_test.cc]: Release the pointer of `data`.

File: src/cmd/sql_cmd.h [hunk 2]
```diff
@@ -282,7 +284,12 @@ void SaveResultSet(::hybridse::sdk::ResultSet *result_set, const std::string &fi
                         if (i != schema->GetColumnCnt()-1) {
                             rowString.append(options->GetDelimiter());
                         } else {
-                            options->GetOfstream() << rowString << std::endl;
+                            if (!first) {
+                                options->GetOfstream() << std::endl;
+                            } else {
+                                first = false;
```
Comment (vagetablechicken) [inline src/cmd/sql_cmd.h]: You'll check first for every column. It's not a good impl. It's better to create a func `GenOneLine`, and the pseudocode is like:
```
fstream << GenOneLine
while Next:
    fstream << std::endl << GenOneLine
```

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: src/cmd/sql_cmd.h (hunk 2 above)
Comment (vagetablechicken) [reply src/cmd/sql_cmd.h]: Maybe refactor later.

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**